manager/src/main/java/com/redhat/service/smartevents/manager/api/user/validators/processors/GatewayConstraintValidator.java

```diff
@@ -0,0 +1,107 @@
+package com.redhat.service.smartevents.manager.api.user.validators.processors;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+import org.hibernate.validator.internal.engine.messageinterpolation.util.InterpolationHelper;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.ActionProviderException;
+import com.redhat.service.smartevents.infra.models.actions.Action;
+import com.redhat.service.smartevents.infra.models.actions.Source;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+import com.redhat.service.smartevents.manager.api.models.requests.ProcessorRequest;
+import com.redhat.service.smartevents.processor.GatewayConfigurator;
+import com.redhat.service.smartevents.processor.actions.ActionValidator;
+
+@ApplicationScoped
+public class GatewayConstraintValidator implements ConstraintValidator<ValidGateway, ProcessorRequest> {
+
+    static final String ACTION_TYPE_MISSING_ERROR = "Action type must be specified";
+    static final String ACTION_TYPE_NOT_RECOGNISED_ERROR = "Action of type '{type}' is not recognised.";
+    static final String ACTION_PARAMETERS_MISSING_ERROR = "Action parameters must be supplied";
+    static final String ACTION_PARAMETERS_NOT_VALID_ERROR = "Parameters for action '{name}' of type '{type}' are not valid.";
+
+    static final String SOURCE_TYPE_MISSING_ERROR = "Source type must be specified";
+    static final String SOURCE_TYPE_NOT_RECOGNISED_ERROR = "Source of type '{type}' is not recognised.";
+    static final String SOURCE_PARAMETERS_MISSING_ERROR = "Source parameters must be supplied";
+    static final String SOURCE_PARAMETERS_NOT_VALID_ERROR = "Parameters for source '{name}' of type '{type}' are not valid.";
+
+    static final String TYPE_PARAM = "type";
+
+    @Inject
+    GatewayConfigurator gatewayConfigurator;
+
+    @Override
+    public boolean isValid(ProcessorRequest value, ConstraintValidatorContext context) {
+        Action action = value.getAction();
+        Source source = value.getSource();
+
+        if (action == null && source == null || action != null && source != null) {
+            return false;
```

Comment:
@kostola I know this is a PoC, but we really need a clear error message here to the user that a Processor must either be a source or a sink.

Reply:
This is one of the shortcuts I took for the first version of the PoC. I will implement it.